dbux-data/src/impl/indexes/ParentTracesInRealContextIndex.js

```diff
@@ -34,9 +34,9 @@ export default class ParentTracesInRealContextIndex extends CollectionIndex {
          */
         added: (contexts) => {
           for (const context of contexts) {
-            const { parentTraceId, contextType } = context;
-            // skip parent trace of virtualContext
-            if (isVirtualContextType(contextType)) {
+            const { parentTraceId, contextId } = context;
+            // skip empty contexts
+            if (this.dp.indexes.traces.byContext.getSize(contextId) === 0) {
```

Comment:
Why allow contexts for which `isVirtualContextType` is `true`?

Reply:
This index is used to find all caller trace of real contexts. We want to register the **first** virtual context(`ResumeAsync`, `ResumeGen`, `ResumeAsyncGen`, but not `Await`), but not the following virtual contexts. The following contexts do not have `parentTraceId` so they will be filtered out this way. And `Await` contexts have no children, so check size of `traces.byContext` index will filter them out.
Maybe the name itself is ambiguous, I'll try to rename them.